oml/datasets/pairs.py

```diff
@@ -1,113 +1,44 @@
-from pathlib import Path
-from typing import Any, Dict, List, Optional, Union
+from typing import Dict, List, Tuple
 
 from torch import Tensor
 
 from oml.const import INDEX_KEY, PAIR_1ST_KEY, PAIR_2ND_KEY
-from oml.datasets.list_dataset import ListDataset, TBBoxes
+from oml.datasets.base import BaseDataset
 from oml.interfaces.datasets import IPairsDataset
-from oml.transforms.images.torchvision import get_normalisation_torch
-from oml.transforms.images.utils import TTransforms
-from oml.utils.images.images import TImReader, imread_pillow
 
 
-class EmbeddingPairsDataset(IPairsDataset):
+class PairsDataset(IPairsDataset):
     """
-    Dataset to iterate over pairs of embeddings.
+    Dataset to iterate over pairs of items.
 
     """
 
     def __init__(
         self,
-        embeddings1: Tensor,
-        embeddings2: Tensor,
+        base_dataset: BaseDataset,
+        pair_ids: List[Tuple[int, int]],
         pair_1st_key: str = PAIR_1ST_KEY,
         pair_2nd_key: str = PAIR_2ND_KEY,
         index_key: str = INDEX_KEY,
     ):
-        """
-
-        Args:
-            embeddings1: The first input embeddings
-            embeddings2: The second input embeddings
-            pair_1st_key: Key to put ``embeddings1`` into the batches
-            pair_2nd_key: Key to put ``embeddings2`` into the batches
-            index_key: Key to put samples' ids into the batches
-
-        """
-        assert embeddings1.shape == embeddings2.shape
-        assert embeddings1.ndim >= 2
+        self.base_dataset = base_dataset
+        self.pair_ids = pair_ids
 
         self.pair_1st_key = pair_1st_key
         self.pair_2nd_key = pair_2nd_key
-        self.index_key = index_key
-
-        self.embeddings1 = embeddings1
-        self.embeddings2 = embeddings2
+        self.index_key: str = index_key
```

Comment:
`self.index_key: str = index_key` -> `self.index_key = index_key` because of `index_key: str = INDEX_KEY` in `__init__`